### DISCUSSION TITLE
Trying to reference a View

### DISCUSSION THREAD

**Discussion by shortsam1** [serenity-is/Serenity#5771] [Help] [answered]
I am trying to reference a view instead of a table in my solution, but I am doing something wrong.  Please help.

![image](https://user-images.githubusercontent.com/47765946/120763553-ae7ba780-c55a-11eb-8cd5-66b7966c8811.png)

`    "Report": {
      "ConnectionString": "Server=XXXXXXXX;Database=Report;User ID=XXXXXXXX;Password=XXXXXXXX",
      "ProviderName": "System.Data.SqlClient"`

`    [ConnectionKey("Report"), Module("AmoAllocation"), TableName("[dbo].[AMO_Specialty_View]")]`

The error I am getting is the following:

**Microsoft.Data.SqlClient.SqlException: 'Invalid object name 'dbo.AMO_Specialty_View'.'**

**Comment by edwardch** [comment, marked as answer]
That looks correct (I have also done the same to reference a view in my xyzRow.cs).

Might be a silly question, but are you sure the connection string hitting the right database? Your database name is "Report", right?

**Reply by shortsam1** [reply]
Yes it is as I can reference the tables in the same database just fine.  When I run sergen, on the report database it does not show any of the 'Views' in the database.

**Reply by edwardch** [reply]
Do you have more than one `appsettings.json` file by any chance? Like `appsettings.Development.json`?

**Reply by shortsam1** [reply]
Yes.  I am running in the dev environment at the moment.  Both my appsettings.json and appsettings.development.json have the following:

appsettings.json
```
`  "Data": {
    "Default": {
      "ConnectionString": "Server=(localdb)\\MSSQLLocalDB;Database=XXXXX;Integrated Security=true",
      "ProviderName": "System.Data.SqlClient"
    },

    "Report": {
      "ConnectionString": "Server=WWWWWWWWW;Database=Report;User ID=WWWWW;Password=WWWWWWWW",
      "ProviderName": "System.Data.SqlClient"
    }`

```
appsettings.development.json
```
`  "Data": {
    "Default": {
      "ConnectionString": "Server=(localdb)\\MSSQLLocalDB;Database=XXXXX;Integrated Security=true",
      "ProviderName": "System.Data.SqlClient"
    },

    "Report": {
      "ConnectionString": "Server=YYYYYYYYY;Database=Report;User ID=YYYYY;Password=YYYYYYYY",
      "ProviderName": "System.Data.SqlClient"
    }`
```

**Reply by edwardch** [reply]
Ok so I see your report database servers are different (assuming xxxx and yyyy are intentionally different).

Last question - are you sure the view is definitely on both servers?

I ask this because I have had an issue with sergen in the past where migrations are run against my databases in appsettings.Development.json, but sergen only looks at the database tables in appsettings.json. So sergen didn't see the new tables that were created from my migrations. That caught me out once or twice before.

**Reply by shortsam1** [reply]
Actually I just commented them out differently so you could see that I took the settings from the different appsettings pages.  They both point to the same server, with the same login and password.

**Reply by edwardch** [reply]
It's hard to say unfortunately without knowing more about how the code has been generated or configured.

It might be that your user in the appsettings doesn't have access to this view, but the user you log into SSMS with _has_ got permission. Do you get this error when trying to list the grid? Or another time?

**Reply by edwardch** [reply]
Hey @shortsam1, checking if you made any progress here?

**Reply by shortsam1** [reply]
Yes.  It turns out the error I was getting was actually to do with linked servers.  I was accessing a table on one db in a form/grid and then trying to read a view that sits on another server from within the same form/grid, so it could not find the view.  The solution was to add the second server as a liked server.  Once I did that it worked well.
